safesocket/safesocket_darwin.go

```diff
@@ -17,23 +20,246 @@ import (
 	"sync"
 	"time"
 
+	"golang.org/x/sys/unix"
 	"tailscale.com/version"
 )
 
 func init() {
 	localTCPPortAndToken = localTCPPortAndTokenDarwin
 }
 
-// localTCPPortAndTokenMacsys returns the localhost TCP port number and auth token
+// SameUserProofTokenLength is the length in bytes of the token used to
+// authenticate safesocket connections
+const (
+	sameUserProofTokenLength = 10
+	SyxExtLibraryDir         = "/Library/Tailscale"
+)
+
+type darwinCredentials struct {
+	mu    sync.Mutex
+	creds struct {
+		token string
+		port  int
+	}
+}
+
+var (
+	// explicit credentials set by SetCredentials
+	dc darwinCredentials
```

Comment:
short variables for short scopes, bigger variable names for bigger (package!) scopes

if you drop the "creds" struct field above and put it here instead, the callers look about the same